library/packages/src/lib/y2packager/license.rb

```diff
@@ -12,40 +12,99 @@
 
 require "yast"
 require "digest"
+require "y2packager/licenses_fetchers"
 
 module Y2Packager
   # Represent a License which could be the same for multiple products.
   #
-  # This class stores the license ID and the traslated content of the license
-  # for different languages.
+  # This class represents a license.
   class License
     DEFAULT_LANG = "en_US".freeze
-    # @return [String] License unique identifier
-    attr_reader :id
 
     # @return [Boolean] whether the license has been accepted or not
     attr_reader :accepted
 
     # @return [Hash<String, String>] language -> content
     attr_reader :translations
 
+    # @return [Y2Packager::LicensesFetchers::Base] License fetcher object
+    attr_reader :fetcher
+
     alias_method :accepted?, :accepted
 
+    class << self
+      # Find a license for a given product
+      #
+      # @param product_name [String]    Product's name
+      # @param source       [:rpm,:url] Source to get the license from. For the time being,
```

Comment:
well, I am not fun of this kind of hidden polymorphysm. Why this method need to know what is source of license? It should find license independently where it came from.

Reply:
I see your point (I do not like it either). But at this point of time, we need to support different uses cases for backward compatibility (like, for instance, retrieving the license elsewhere in the code). In the future we should be able to determine which fetcher/provider to use automatically.

Additionaly, we will drop :url for the now.